src/sentry/conf/server.py

```diff
@@ -32,20 +41,24 @@ def gettext_noop(s: str) -> str:
 socket.setdefaulttimeout(5)
 
 
+T = TypeVar("T")
+TypeParser = TypeVar("TypeParser", bound=Type)
+
+
 @overload
 def env(key: str) -> str:
     ...
 
 
 @overload
-def env(key: str, default: T, type: Callable[[Any], T] | None = None) -> T:
+def env(key: str, default: T, type: Optional[TypeParser] = None) -> T:
     ...
 
 
 def env(
     key: str,
     default: str | T = "",
-    type: Optional[Callable[[Any], T]] = None,
+    type: Optional[TypeParser] = None,
 ) -> T:
```

Comment:
This new function signature only mentions the typevar `TypeParser` once, which makes it unable to constrain any calls to this function. If the intent is to just constrain T to `sentry.utils.types.Type` i'd do this:

```
from sentry.utils.types import Type
T_Type = TypeVar("T_Type", bound=Type)

def env(
    key: str,
    default: T_Type | str = "",
    type: T_Type | None = None,
) -> T_Type:

```